pos_payment_terminal/static/src/js/pos_payment_terminal.js

```diff
@@ -57,7 +57,32 @@ odoo.define('pos_payment_terminal.pos_payment_terminal', function (require) {
             data.amount = line.get_amount();
             data.payment_mode = line.cashregister.journal.payment_mode;
         },
+        handle_terminal_response: function(line_id, response){
+            // This is intended to entrich payment line with
+            // useful values (status, transaction_id, ...)
+            if (response === undefined){
+                return;
+            }
+            var self = this;
+            var order = self.pos.get_order()
+            var line = order.get_paymentline(line_id)
+            var paymentwidget = self.pos.chrome.screens.payment;
+            if ('transaction_id' in response){
+                line.transaction_id = response['transaction_id']
+            }
```

Comment:
You can simplify by writing `line.transaction_id = response.transaction_id`, no need to test for the field presence.